score-server/src/main/java/bio/overture/score/server/config/SecurityConfig.java

```diff
@@ -163,4 +130,77 @@ private boolean useJwt(HttpServletRequest request) {
     }
     return true;
   }
+
+  @Bean
+  public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+    http.authorizeHttpRequests(
+            authorize ->
+                authorize
+                    .antMatchers("/isAlive")
+                    .permitAll()
+                    .antMatchers("/studies/**")
+                    .permitAll()
+                    .antMatchers("/upload/**")
+                    .permitAll()
+                    .antMatchers("/entities/**")
+                    .permitAll()
+                    .antMatchers("/export/**")
+                    .permitAll()
+                    .antMatchers("/schemas/**")
+                    .permitAll()
+                    .antMatchers(swaggerConfig.getAlternateSwaggerUrl())
+                    .permitAll()
+                    .antMatchers(
+                        "/swagger**",
+                        "/swagger-ui.html",
+                        "/swagger-ui**",
+                        "/swagger-resources/**",
+                        "/v2/api-docs/**",
+                        "/webjars/**")
+                    .permitAll()
+                    .anyRequest()
+                    .authenticated())
+        .oauth2ResourceServer(
+            oauth2 -> oauth2.authenticationManagerResolver(tokenAuthenticationManagerResolver()));
+
+    return http.build();
+  }
+
+  @Bean
+  public AuthenticationManagerResolver<HttpServletRequest> tokenAuthenticationManagerResolver() {
+    AuthenticationManager jwtManager =
+        new ProviderManager(new JwtAuthenticationProvider(jwtDecoder));
+
+    AuthenticationManager authzOpaqueManager =
+        new ProviderManager(new OpaqueTokenAuthenticationProvider(authzTokenIntrospector));
+
+    return (request) -> {
+      if (isAuthZRequest(request)) {
+        return authzOpaqueManager;
+      } else {
+        return jwtManager;
+      }
+    };
+  }
+
+  private boolean isAuthZRequest(HttpServletRequest request) {
+    // Option 1: Check for custom header
+    String providerHeader = request.getHeader("X-Auth-Provider");
+    if (providerHeader != null && providerHeader.equalsIgnoreCase("authz")) {
+      return true;
+    }
+
+    // Option 2: Detect based on token format
+    String authHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
+    if (authHeader != null && authHeader.startsWith("Bearer ")) {
+      String token = authHeader.substring(7);
+      return !isJwtToken(token); // Not a JWT => probably opaque AuthZ token
+    }
+
+    return false;
+  }
```

Comment:
The file-transfer application is responsible for choosing which authorization mechanism is in use, not requester. Selecting which authorization provider is used is controlled via the `auth.server.provider` environment property, same as it was with `file-manager`.

We do not want to let the requester select their provider. Let's remove this function.

Instead, lets do what we did with the `file-manager` and apply auth rules based on the configured `provider`.